packages/salmon/__test__/publish.test.ts

```diff
@@ -0,0 +1,227 @@
+import waitForExpect from 'wait-for-expect'
+import { JsonRpcClient } from '@defichain/jellyfish-api-jsonrpc'
+import { RegTestGenesisKeys } from '@defichain/jellyfish-network'
+import { WhaleMasternodeRegTestContainer } from '@defichain/salmon-testing'
+import { SalmonWallet } from '@defichain/salmon-wallet'
+import { BigNumber } from 'bignumber.js'
+import { AssetPrice, Salmon, WhaleApiClient } from '../src'
+import coingecko from '@defichain/oracle-adapters/src/coingecko'
+
+const container = new WhaleMasternodeRegTestContainer()
+const address = RegTestGenesisKeys[RegTestGenesisKeys.length - 1].operator
+let client: JsonRpcClient
+let whaleApiClient: WhaleApiClient
+let oracleId: string
+
+beforeEach(async () => {
```

Comment:
Gut feeling, the test work without re-setup on each. Worth trying just `beforeAll`

Reply:
I really clone the code and test. Ivan is correct